### PR TITLE
For SG-19625: gracefully report errors with older PySide2 versions fo…

### PR DESCRIPTION
**Pull request by patrick-hubert-adsk**
…r missing methods/classes

This has proven necessary to properly support Flame. I suspect this will be helpful for other packages.

### CODE REVIEW

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py [hunk 1]
```diff
@@ -164,6 +165,12 @@ def __init__(self, window_title="Web Login", qt_modules=None):
             raise SsoSaml2MissingQtWebKit(
                 "The QtWebKit or QtWebEngineWidgets modules are unavailable"
             )
+        if QtWebEngineWidgets and not hasattr(
```
Comment (jfboismenu) [inline python/tank/authentication/sso_saml2/core/sso_saml2_core.py:168]: Could you add a comment explaining when this happens, like likely versions of PySide this is an issue?

**Review by jfboismenu — APPROVED** [review]
Some clarification about the check you're making would be appreciated, but aside from that it looks all good.

Thank you for the quick turnaround! Flame clients will be able to stay up to date!

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py (hunk 1 above)
Comment (patrick-hubert-adsk) [reply python/tank/authentication/sso_saml2/core/sso_saml2_core.py:180]: Added a long(er) comment explaining the issue(s).

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py [hunk 2]
```diff
@@ -165,6 +165,18 @@ def __init__(self, window_title="Web Login", qt_modules=None):
             raise SsoSaml2MissingQtWebKit(
                 "The QtWebKit or QtWebEngineWidgets modules are unavailable"
             )
+
+        # If PySide2 is being used, we need to make  extra checks to ensure
+        # that needed components are indeed presentself.
```
Comment (jfboismenu) [inline python/tank/authentication/sso_saml2/core/sso_saml2_core.py:170]:
```suggestion
        # that needed components are indeed present.
```

**Review by jfboismenu — APPROVED** [review]
One minor typo, otherwise...

image
# work!

File: python/tank/authentication/sso_saml2/core/sso_saml2_core.py (hunk 2 above)
Comment (patrick-hubert-adsk) [reply python/tank/authentication/sso_saml2/core/sso_saml2_core.py]: Damned IntelliSense code snippet...

**Review by jfboismenu — APPROVED** [review]

**Comment by patrick-hubert-adsk** [comment]
The Azure builds were broken at the time of my last commit, so the check could not be successful.
Now that they have been fixed, I will merge my branch in master.

**Pull request merged**